### PR TITLE
Add possibility to use multiple dependencies in widgets with the same source field

### PR DESCRIPTION
**Pull request by renttek**
### Description (*)

Add the possibility to use multiple values for the same parameter-name in widgets parameter-dependencies.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<widgets xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Widget:etc/widget.xsd">
    <widget id="simple_test" class="My\Module\Widget\SimpleTest">
        <label>Simple test</label>
        <description>This won't work</description>
        <parameters>
            <parameter name="test_1" xsi:type="select" visible="true" source_model="Magento\Config\Model\Config\Source\Yesnocustom">
                <label>Foo? Bar? Baz?</label>
            </parameter>
            <parameter name="test_2" xsi:type="text" required="true" visible="true">
                <label>Useless text</label>
                <depends>
                    <parameter name="test_1" value="1"/>
                    <parameter name="test_1" value="2" />
                </depends>
                <value>5</value>
            </parameter>
        </parameters>
    </widget>
</widgets>
```

When trying to use this widget in a content page or block, the ajax-call to render the fields would return a HTTP 500 error.
This is caused by 2 pieces of code:

1. An array to string conversion in `Magento\Backend\Block\Widget\Form\Element\Dependence::addFieldDependence`:

```php
    public function addFieldDependence($fieldName, $fieldNameFrom, $refField)
    {
        if (!is_object($refField)) {
            /** @var $refField \Magento\Config\Model\Config\Structure\Element\Dependency\Field */
            $refField = $this->_fieldFactory->create(
                ['fieldData' => ['value' => (string)$refField], 'fieldPrefix' => ''] // <- error will happen here
            );
        }
        $this->_depends[$fieldName][$fieldNameFrom] = $refField;
        return $this;
    }
```

2. Always wrapping the values in an array in `Magento\Config\Model\Config\Structure\Element\Dependency\Field::__construct`

```php
    public function __construct(array $fieldData = [], $fieldPrefix = "")
    {
        if (isset($fieldData['separator'])) {
            $this->_values = explode($fieldData['separator'], $fieldData['value']);
        } else {
            $this->_values = [isset($fieldData['value']) ? $fieldData['value'] : '']; // <- this won't work with an array as value
        }
        $fieldId = $fieldPrefix . (isset(
            $fieldData['dependPath']
        ) && is_array(
            $fieldData['dependPath']
        ) ? array_pop(
            $fieldData['dependPath']
        ) : '');
        $fieldData['dependPath'][] = $fieldId;
        $this->_id = implode('_', $fieldData['dependPath']);
        $this->_isNegative = isset($fieldData['negative']) && $fieldData['negative'];
    }
```

This PR patches both pieces of code, while making sure, that existing code will function as before.

### Fixed Issues (if relevant)
1. #31265

### Manual testing scenarios (*)
1. Create a module with a simple widget (with the given `widget.xml` from above)
2. Try to add this widget to a cms page
3. Widget form won't render properly

### Contribution checklist (*)
- [ ] Pull request has a meaningful description of its purpose
- [ ] All commits are accompanied by meaningful commit messages
- [ ] All new or changed code is covered with unit/integration tests (if applicable)
- [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lbajsarowicz** [comment]
@magento run all tests

File: app/code/Magento/Backend/Test/Unit/Block/Widget/Form/Element/DependenceTest.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+namespace Magento\Backend\Test\Unit\Block\Widget\Form\Element;
+
+use Magento\Backend\Block\Context;
+use Magento\Backend\Block\Widget\Form\Element\Dependence;
+use Magento\Config\Model\Config\Structure\Element\Dependency\FieldFactory;
+use Magento\Framework\Json\EncoderInterface;
+use Magento\Framework\View\Helper\SecureHtmlRenderer;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use stdClass;
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Unit/Block/Widget/Form/Element/DependenceTest.php]: You don't need to import the short class names.

File: app/code/Magento/Backend/Test/Unit/Block/Widget/Form/Element/DependenceTest.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     * @var Dependence
+     */
+    private $dependence;
+
+    protected function setUp(): void
+    {
+        $this->contextMock            = $this->createMock(Context::class);
+        $this->jsonEncoderMock        = $this->createMock(EncoderInterface::class);
+        $this->fieldFactoryMock       = $this->createMock(FieldFactory::class);
+        $this->secureHtmlRendererMock = $this->createMock(SecureHtmlRenderer::class);
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Unit/Block/Widget/Form/Element/DependenceTest.php]: Could you lint the code **without** argument alignment.

File: app/code/Magento/Backend/Test/Unit/Block/Widget/Form/Element/DependenceTest.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            [],
+            $this->secureHtmlRendererMock
+        );
+    }
+
+    /**
+     * @param string|array $value
+     * @param array        $expectedFieldValues
+     *
+     * @dataProvider fieldDependenceValuesDataProvider
```
Comment (lbajsarowicz) [inline app/code/Magento/Backend/Test/Unit/Block/Widget/Form/Element/DependenceTest.php]: Also lint the code without arguments alignment

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
🟡  I don't see any critical issues. For sure - due to the change of parameter type - there will be necessary manual approval from Community Engineering Team.

@renttek Could you approve the CLA?

**Comment by renttek** [comment]
I already signed the CLA, but the check seemed to not detect it until I just updated the PR.

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by andrewbess** [comment]
This PR is merged.
Please reopen if it needed

**Pull request closed without merging**